Source/Core/Form.swift

```diff
@@ -213,22 +213,66 @@ extension Form : RangeReplaceableCollection {
         }
     }
 
-    public func replaceSubrange<C: Collection>(_ subRange: Range<Int>, with newElements: C) where C.Iterator.Element == Section {
+    public func replaceSubrange<C: Collection>(
+        _ subRange: Range<Int>,
+        with newElements: C
+    ) where C.Iterator.Element == Section {
         for i in subRange.lowerBound..<subRange.upperBound {
             if let section = kvoWrapper.sections.object(at: i) as? Section {
                 section.willBeRemovedFromForm()
                 kvoWrapper._allSections.remove(at: kvoWrapper._allSections.firstIndex(of: section)!)
             }
         }
-        kvoWrapper.sections.replaceObjects(in: NSRange(location: subRange.lowerBound, length: subRange.upperBound - subRange.lowerBound),
-                                           withObjectsFrom: newElements.map { $0 })
+        kvoWrapper.sections.replaceObjects(
+            in: NSRange(location: subRange.lowerBound, length: subRange.upperBound - subRange.lowerBound),
+            withObjectsFrom: newElements.map { $0 }
+        )
         kvoWrapper._allSections.insert(contentsOf: newElements, at: indexForInsertion(at: subRange.lowerBound))
 
         for section in newElements {
             section.wasAddedTo(form: self)
         }
     }
 
+    public func replaceSubrangeInAllSections<C: Collection>(
+        _ subRange: Range<Int>,
+        with newElements: C
+    ) where C.Iterator.Element == Section {
+        // Keep a reference to all removed sections in order to create a subrange for the visible section replacement
+        var removedSections = [Section]()
+
+        // Remove subrange in all sections
+        for i in subRange.reversed() where kvoWrapper._allSections.count > i {
+            let removed = kvoWrapper._allSections.remove(at: i)
+            removed.willBeRemovedFromForm()
+            removedSections.insert(removed, at: 0)
+        }
+
+        let removedVisibleSections = removedSections.filter { !$0.isHidden }
+
+        // Get indices of first and last section which is not hidden...
+        let firstVisible = kvoWrapper.sections.contains(removedVisibleSections.first) ? kvoWrapper.sections.index(of: removedVisibleSections.first) : 0
```

Comment:
I know these sections should be in the kvoWrapper.sections array but the fallback of 0 does not look good in case something does not behave as expected. You could create a range like (X, 0) which would be invalid or (0, X) which  would remove more sections than wanted. What about an if let?

Reply:
Thinking about this and how to solve it took me way longer than I thought it would. 😁
The range which would have been created should have never been invalid (in terms of crash-safety), because `last >= first` should always be true in case of indices. 
Still the fallback of `0` doesn't make sense,  as you have already mentioned. Some potential problems would have been:
- Replacing only hidden sections at the end of the array with a set of visible sections
- Replacing only hidden sections *in between* two visible sections

I first tried to come up with a solution which handles all of the problematic scenarios. But then I suddenly had a brainwave, where I remembered you told me that all hidden sections are going to be removed anyway.

So the most easy solution is probably just to replace all visible sections by `_allSections` after processing it. This works for us. I hope this makes sense.